webapi/src/main/scala/org/knora/webapi/util/PermissionUtilV1.scala

```diff
@@ -316,106 +315,36 @@ object PermissionUtilV1 {
-                                                     predicateIri: IRI,
-                                                     valueProps: ValueProps,
-                                                     subjectProject: Option[IRI],
-                                                     userProfile: UserProfileV1): Option[Int] = {
-        if (predicateIri == OntologyConstants.KnoraBase.HasStandoffLinkTo) {
-            Some(permissionsToV1PermissionCodes(OntologyConstants.KnoraBase.ViewPermission))
-        } else {
-            getUserPermissionV1WithValueProps(
-                subjectIri = linkValueIri,
-                valueProps = valueProps,
-                subjectProject = subjectProject,
-                userProfile = userProfile
-            )
-        }
-    }
-
-    /**
-      * Determines the permissions that a user has on a `knora-base:LinkValue`, and returns a permissions code in Knora API v1 format.
-      * If the `rdf:predicate` of the `LinkValue` is [[OntologyConstants.KnoraBase.HasStandoffLinkTo]], this method always returns
-      * view permission. Otherwise, it returns the value returned by [[getUserPermissionV1WithValueProps]].
-      *
-      * @param linkValueIri               the IRI of the link value.
-      * @param predicateIri               the IRI of the link value's `rdf:predicate`.
-      * @param linkValueCreator             the IRI of the link value's creator.
-      * @param containingResourceProject  the IRI of the project of the resource that contains the link value.
-      * @param linkValuePermissionLiteral the literal object of the link value's `knora-base:hasPermissions` predicate.
-      * @param userProfile                the profile of the user making the request.
-      * @return a code representing the user's permission level on the value.
-      */
-    def getUserPermissionOnLinkValueV1(linkValueIri: IRI,
-                                       predicateIri: IRI,
-                                       linkValueCreator: IRI,
-                                       containingResourceProject: IRI,
-                                       linkValuePermissionLiteral: Option[String],
-                                       userProfile: UserProfileV1): Option[Int] = {
-        if (predicateIri == OntologyConstants.KnoraBase.HasStandoffLinkTo) {
-            Some(permissionsToV1PermissionCodes(OntologyConstants.KnoraBase.ViewPermission))
-        } else {
-            getUserPermissionV1(
-                subjectIri = linkValueIri,
-                subjectCreator = linkValueCreator,
-                subjectProject = containingResourceProject,
-                subjectPermissionLiteral = linkValuePermissionLiteral,
-                userProfile = userProfile
-            )
-        }
-    }
-
-    /**
       * Parses the literal object of the predicate `knora-base:hasPermissions`.
       *
-      * @param maybePermissionListStr the literal to parse.
+      * @param permissionListStr the literal to parse.
```

Comment:
In the ontology, `kb:hasPermission` has a `:objectDatatypeConstraint` of `xsd:string`. Could this be made more specific? You do not just assume any string, but a specific string format.

In XML-schema, one can provide a pattern. 

This is an example taken from the *Meditationes* constraining attribute values to a certain pattern:
```
<xs:pattern value="(\s?underline|\s?center|\s?strike|\s?above|\s?marginal|\s?below)+"/>
```